### DISCUSSION TITLE
Custom search appears to break people contact card click

### DISCUSSION THREAD

**Discussion by cluber22** [microsoft-search/pnp-modern-search#3376] [Q&A] [answered]
I've implemented a custom people search on my SharePoint site and that has now broken the people search when people start typing in the the search box and click the suggested results (see screenshot). The search box i'm talking about is the default one at the top middle of SharePoint Online.

Clicking from the contact card itself seems to try and pass the following query string to the search page.

`Search.aspx?csf=1&web=1&e=5dgJqe&q=&k=&pp=e5bb4bac-0768-43e1-af00-cba33475641c@6b0ff425-e5e2-4239-bd8b-91ba02b7940a|david.smith@domain.com#`

I think the problem is that the custom search doesn't know how to handle the **pp= parameter** and therefore know which email address to pass to the search. The result been that it just lists all users and not the person that you clicked in the search bar.

How can I get this to work?

Thank you

![sharepoint-search-person](https://github.com/microsoft-search/pnp-modern-search/assets/3111945/1aa72b21-bcc3-48e3-bd11-a2e59a5b024d)

![pnp-1-4](https://github.com/microsoft-search/pnp-modern-search/assets/3111945/fcf8b62b-aaa4-4e9c-9307-0dfd1bc54abd)
![pnp-2-4](https://github.com/microsoft-search/pnp-modern-search/assets/3111945/216f9914-f323-4f85-8101-8637b8a7699d)
![pnp-3-4](https://github.com/microsoft-search/pnp-modern-search/assets/3111945/d4c9813d-8c76-4fb8-9b92-d24f51bd63e3)
![pnp-4-4](https://github.com/microsoft-search/pnp-modern-search/assets/3111945/b4ed0dee-418c-4f94-ad62-0cf3ae6f9a63)

**Comment by kasperbolarsen** [comment]
Hi @cluber22 

It is VERY difficult for us to offer any assistance when we know nothing about your setup.

Please provide the version number of the version you are using, and a screen shot of each of the 4 pages for at least the Search Box ( and the Results web part too if possible)

**Reply by cluber22** [reply]
Images have been added.
The search box is the default one at the top of SharePoint, it works fine if I just search but not when I click the contact card.
I also notice that my version is quite a bit behind current. 4.5.3 compared to 4.9.3

**Comment by cluber22** [comment]
UPDATE - I've updated to 4.9.3 and the issue persists.

Is it meant to break the out of the box functionality of the SharePoint Search suggestions?

**Reply by kasperbolarsen** [reply]
I am pretty surprised that there is any kind of relation between the Out Of The Box Microsoft Search box in the suite bar and PnP Modern Search Web parts.
A few questions: what is the search scope for the Microsoft Search box, site, Hub or Tenant? ( get-pnpsearchsettings)
What does the custom Result source looks like?  
does changing to localPeopleResults do anything?

**Reply by cluber22** [reply]
**get-pnpsearchsettings**
Classic Search Center URL                   :
Site Search Scope                           : DefaultScope
Site collection search box visibility       : Inherit
Site search box visibility                  : Inherit
Site collection search box placeholder text :
Site search box placeholder text            : Search this site, phonebook, documents...

**Site Settings -> Search Settings**
Enter a Search Center URL: https://mydomain.sharepoint.com/SitePages/Search.aspx
Which search results page should queries be sent to? Turn on the drop-down menu inside the search box, and use the first Search Navigation node as the destination results page.
Everything: https://mydomain.sharepoint.com/SitePages/Search.aspx
People: https://mydomain.sharepoint.com/search/Pages/peopleresults.aspx
Conversations: https://mydomain.sharepoint.com/search/Pages/conversationresults.aspx

I will try the localPeopleResults result source, I am currently using a custom one which is Protocol="Local SharePoint" and Type="People Search Results"

**Comment by till-llit** [comment]
hi @cluber22 : we had the same problem. our solution was to write a custom SPFX Webpart (called "redirect webpart") with the only purpose to watch the URL Parameter "pp" and redirect the  user to the delve / profile page of the user.

**Reply by cluber22** [reply]
Hi @till-llit thanks for coming back to me on this. I thought I was doing something wrong and I am struggling on a way forward with this. Could you potentially share some details on how I can achieve this technically. I will get up to speed on creating a webpart but specifically the change that's needed around the Search bar.

**Reply by till-llit** [reply]
hi @cluber22, im not allowed to share the source code - but every SPFX developer should be able to develop this webpart in half a day.

the code is just like "check if parameter is in url -> if yes: inject a `<a>`  element in the DOM and click it.

**Reply by cluber22** [reply]
I appreciate you can't share company source code @till-llit but can you give me a bit of a leg up or reference as to how I'm meant to override the main search bar. I've been reading into creating a web part in general but can't find anything regarding the main search.
Thank you

**Reply by till-llit** [reply]
the idea behind is not to override the search bar. just let the user click on the people result - this will navigate the user to your custom search result page, but with the ?pp= parameter attached to the url. and here is where your "invisible" redirect webpart is placed. just put it somewhere on the page. the webpart will watch the url for the pp parameter and redirect the user from your custom search result page to a profil display page like delve.

**Reply by cluber22** [reply]
Thank you I think I've with you now, much appreciated. Do you think this is a bug somewhere as this seems like a horrible workaround.

**Reply by till-llit** [reply, marked as answer]
if it is a "bug" - its not a bug caused by this solution here (pnp modern search) but by the search solution from Microsoft.

SharePoint is providing a way to redirect search box queries to a custom page and just appends the search terms with k= and q= url parameters. Additionally they provide search suggestions in the flyout (sites and documents) with a direct link to the site and document. But for some - for me unknown - reason they decided to provide links to people suggestions with a undocumented and nowhere else used "pp=" parameter. 

I think it would be helpful for others if PnP Modern Search provides a way to handle this pp= parameter because everybody will run into this problem who is redirecting the SharePoint Search Box to a custom PnP Modern Search result page, but it's not a bug on their side.

**Reply by cluber22** [reply]
Went with this approach

```
private ReplacePpInQueryString(): void {
    const url = new URL(window.location.href);
    const searchParams = new URLSearchParams(url.search);
    const pp = searchParams.get("pp");
    // if contains pp= then we need to take this data and feed it into k= and q=
    if(pp != null)
    {
      // get UPN from pp=
      let result = pp.split("|");
      let UPN = result[1];
      const redirectURL = "SitePages/Search.aspx?q=" + UPN + "&k=" + UPN;
      window.location.href = "https://mydomain.sharepoint.com/" + redirectURL;
    }
}
```
